yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -142,6 +145,99 @@ class YarnSparkHadoopUtil extends SparkHadoopUtil {
     val containerIdString = System.getenv(ApplicationConstants.Environment.CONTAINER_ID.name())
     ConverterUtils.toContainerId(containerIdString)
   }
+
+  /**
+   * Obtains token for the Hive metastore, using the current user as the principal.
+   * Some exceptions are caught and downgraded to a log message.
+   * @param conf hadoop configuration; the Hive configuration will be based on this
+   * @return a token, or `None` if there's no need for a token (no metastore URI or principal
+   *         in the config), or if a binding exception was caught and downgraded.
+   */
+  def obtainTokenForHiveMetastore(conf: Configuration): Option[Token[DelegationTokenIdentifier]] = {
+    try {
+      obtainTokenForHiveMetastoreInner(conf, UserGroupInformation.getCurrentUser().getUserName)
+    } catch {
+      case e: Exception => {
+        handleTokenIntrospectionFailure("Hive", e)
+        None
+      }
+    }
+  }
+
+  /**
+   * Handle failures to obtain a token through introspection. Failures to load the class are
+   * not treated as errors: anything else is.
+   * @param service service name for error messages
+   * @param thrown exception caught
+   * @throws Exception if the `thrown` exception isn't one that is to be ignored
+   */
+  private[yarn] def handleTokenIntrospectionFailure(service: String, thrown: Throwable): Unit = {
+    thrown match {
+      case e: ClassNotFoundException =>
+        logInfo(s"$service class not found $e")
+        logDebug("Hive Class not found", e)
+      case t: Throwable => {
```

Comment:
I don't follow. You're catching an exception, logging it, and re-throwing it, which causes the exception to show up twice in the process output.

Instead, you can just delete your code, and let the exception propagate naturally. It will show up in the output the same way.